datafusion/core/src/physical_plan/hash_utils.rs

```diff
@@ -62,29 +62,23 @@ fn hash_decimal128<'a>(
     if array.null_count() == 0 {
         if mul_col {
             for (i, hash) in hashes_buffer.iter_mut().enumerate() {
```

Comment:
I think we can we use `hash_array` here now instead?

Reply:
Potentially, I think I would rather keep this as a simple upgrade PR, and address any simplifications that fall out of the new decimal abstractions as a follow up